FreeRTOS/FreeRTOS
# Add SMP interrupt wait critical secton on target test

Add SMP interrupt wait critical secton on target test

Description
-----------
Adding the on target test to verify that
> If a task is interrupted while it is waiting to enter a critical section, it shall relinquish the core instead of continuing to wait to enter critical section.

Test Steps
-----------
Test result
```
./FreeRTOS/FreeRTOS/Test/Target/tests/smp/interrupt_wait_critical/interrupt_wait_critical.c:324:Test_InterruptWaitCritical:PASS

-----------------------
1 Tests 0 Failures 0 Ignored
OK
```

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
